### ISSUE TITLE
improve performance of partial resource application

### ISSUE THREAD

**Issue by sanderr** [inmanta/inmanta-core#4743]
The `put_partial` endpoint currently applies the partial by fetching all resources for the base version from the database, then calculating the diff and storing all resources again. The performance could be greatly improved by instead doing a single query to copy all resources that are not part of the current export's resource sets or deleted sets. The diff and consequent insert would then be contained to the resource sets for this call. This might require getting rid of the resource version id field in the DB and only keep the version and ID separate.

Ticket based on PR suggestion [here](https://github.com/inmanta/inmanta-core/pull/4738#discussion_r955047504).

Don't merge before September release!
